build.rs

```diff
@@ -30,18 +30,91 @@ use std::{
 mod env {
     use std::ffi::OsString;
 
-    /// Read an environment variable and tell Cargo that we depend on it.
+    enum EnvVarTy {
+        RerunIfChanged,
+        SetByCargo,
+    }
+
+    pub struct EnvVar {
+        name: &'static str,
+        ty: EnvVarTy,
+    }
+
+    /// Read an environment variable and optionally tell Cargo that we depend on it.
     ///
-    /// The name is static since we intend to only read a static set of environment
+    /// The env var is static since we intend to only read a static set of environment
     /// variables.
-    pub fn var_os(name: &'static str) -> Option<OsString> {
-        println!("cargo:rerun-if-env-changed={}", name);
-        std::env::var_os(name)
+    pub fn var_os(env_var: &'static EnvVar) -> Option<OsString> {
+        if let EnvVarTy::SetByCargo = env_var.ty {
```

Comment:
1. In this codebase we use `match` instead of `if let` for matching enums other than `Option`.
2. Isn't this the opposite of what you want? I think if the env var. is set by cargo, then we DON'T want to print "cargo::rerun-if-env-changed=...", right?

Reply:
1. Fixed.
2. Thanks for catching that! It was the first thing I did this morning, should've waited until I finished my coffee ;-)